mne/beamformer/_lcmv.py

```diff
@@ -73,37 +74,105 @@ def _check_one_ch_type(info, picks, noise_cov):
         [_contains_ch_type(info_pick, tt) for tt in ('mag', 'grad', 'eeg')]
     if sum(ch_types) > 1 and noise_cov is None:
         raise ValueError('Source reconstruction with several sensor types '
-                         'requires a set of noise covariance matrices to be '
+                         'requires a noise covariance matrix to be '
                          'able to apply whitening.')
 
 
+def _pick_channels_spatial_filter(ch_names, spat_filt):
+    """Return data channel indices to be used with spatial filter.
+
+    Unlike ``pick_channels``, this respects the order of ch_names.
+    """
+    sel = list()
+    for name in spat_filt['ch_names']:
+        try:
+            sel.append(ch_names.index(name))
+        except ValueError:
+            raise ValueError('The inverse operator was computed with '
```

Comment:
essentially replace the try/except with a safe membership testing ```ch_name in ch_names``` as Alex suggests.